compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -363,6 +363,7 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
         case Inlined(call, bindings, expansion) if !call.isEmpty =>
           val pos = call.sourcePos
           CrossVersionChecks.checkExperimentalRef(call.symbol, pos)
+          super.transform(call)
```

Comment:
We still have the issue of the exponential blowup.

Reply:
```suggestion
         transform(call)
```

Reply:
By doing this recursion we will always enable `ctx.compilationUnit.needsInlining = true`.

* https://github.com/lampepfl/dotty/blob/main/compiler/src/dotty/tools/dotc/transform/PostTyper.scala#L287
* https://github.com/lampepfl/dotty/blob/main/compiler/src/dotty/tools/dotc/transform/PostTyper.scala#L295
* https://github.com/lampepfl/dotty/blob/main/compiler/src/dotty/tools/dotc/transform/PostTyper.scala#L348

This is not ideal if the file only contains transparent inline calls.